pandas/plotting/_matplotlib/core.py

```diff
@@ -810,7 +816,15 @@ def match_labels(data, e):
             err_shape = err.shape
 
             # asymmetrical error bars
-            if err.ndim == 3:
+            if isinstance(self.data, ABCSeries) and err_shape[0] == 2:
```

Comment:
Can you just have one branch that works generically for Series / DataFrame?

Reply:
I don't think it's possible or useful. The error message is similar, but we perform some additional actions for the Series (adding a dimension) and updating the err_shape.

Reply:
OK. @jreback feel free to merge